indico/htdocs/js/indico/jquery/categorypicker.js

```diff
@@ -0,0 +1,332 @@
+                }
+            });
+        },
+
+        _buildBreadcrumbs: function(path, clickable) {
+            var self = this;
+            var $breadcrumbs = $('<ul>', {class: 'breadcrumbs'});
+            var tag = clickable? '<a>' : '<span>';
+
+            _.each(path, function(category) {
+                var $item = $(tag, {
+                    text: category.title
+                });
+                if (clickable) {
+                    $item.attr('href', '');
+                    $item.attr('title', $T.gettext("Go to: {0}".format(category.title)));
+                }
+                self._bindGoToCategoryOnClick($item, category.id);
+                $breadcrumbs.append($('<li>').append($item));
+            });
+
+            return $breadcrumbs;
+        },
+
+        _buildCategory: function(category, isSubcategory, withBreadcrums, clickableBreadcrumbs) {
+            var self = this;
+            var tag = isSubcategory? '<li>' : '<div>';
+            var itemClass = isSubcategory? 'subcategory' : 'current-category';
+
+            var $category = $(tag, {
+                class: 'item ' + itemClass,
+                id: 'category-' + category.id
+            });
+
+            var $categoryTitle = $('<div>', {class: 'title-wrapper'});
+            $categoryTitle.append($('<span>', {class: 'title',text: category.title}));
+            if (withBreadcrums) {
+                $categoryTitle.append(self._buildBreadcrumbs(category.path, clickableBreadcrumbs));
+            }
+
+            $category.append($('<div>', {class: 'icon-wrapper'}));
+            $category.append($categoryTitle);
+            $category.append(self._buildSidePanel(category, !isSubcategory));
+            return $category;
+        },
+
+        _buildCurrentCategory: function(category) {
+            var self = this;
+            return self._buildCategory(category, false, true, true);
+        },
+
+        _buildSubcategory: function(category, withBreadcrums) {
+            var self = this;
+            var $subcategory = self._buildCategory(category, true, withBreadcrums);
+
+            if (category.can_access) {
+                self._bindGoToCategoryOnClick($subcategory, category.id);
+                $subcategory.addClass('can-access');
+            }
+
+            return $subcategory;
+        },
+
+        _buildSidePanel: function(category, withGoToParent) {
+            var self = this;
+            var $buttonWrapper = $('<div>', {class: 'button-wrapper'});
+
+            $buttonWrapper.append($('<div>').append($('<span>', {
+                class: 'action-button',
+                text: self.options.actionButtonText
+            }).on('click', function(evt) {
+                evt.stopPropagation();
+                self._onAction(category);
+            })));
+
+            if (withGoToParent && category.path && category.path.length) {
+                var parent = _.last(category.path);
+                $buttonWrapper.append($('<a>', {
```

Comment:
would be more readable if you assigned it first, then bound the events and then appended it to the wrapper.

or simply use `$('<a>', {..}).on(...).appendTo($buttonWrapper);` to avoid the call around everything